packer/run_interfaces.go

```diff
@@ -15,6 +15,7 @@ type BuildGetter interface {
 	// builders.
 	// TODO(azr): rename to builder starter ?
 	GetBuilds(GetBuildsOptions) ([]Build, hcl.Diagnostics)
+	Validate(GetBuildsOptions) hcl.Diagnostics
```

Comment:
For validation my original idea was to use these :

https://github.com/hashicorp/packer/blob/ee5635722bd8c34cb114c3cbfe80eafcac78e16a/packer/run_interfaces.go#L32-L54